README.md

```diff
@@ -108,43 +108,86 @@ See [CONFIG.md](CONFIG.md) for details.
-Because predicates run on the server side, the server should be able to reason about your objects. That is why you need to implement serialization on the server side.
+Note that if the object is not instance of any explicit type, Node.js client uses `JSON Serialization`. The default behaviour can be changed by using a `Custom Serialization`.
 
-The same applies to MapStore. The server should be able to deserialize your objects in order to store them in MapStore.
 
-Regarding arrays in a serializable object, you can use methods like `writeIntArray` if the array is of a primitive type.
 
-If you have nested objects, these nested objects also need to be serializable. Register the serializers for nested objects and the method `writeObject` will not have any problem with finding a suitable serializer for and writing/reading the nested object.
+## 1. IdentifiedDataSerializable Serialization
+For a faster serialization of objects, Hazelcast recommends to implement IdentifiedDataSerializable interface.
 
-If you have arrays of custom objects, you can serialize/deserialize them like the following:
+Here is an example of a class implementing IdentifiedDataSerializable interface:
 
 ```javascript
-writeData(dataOutput) {
-    ...
-    dataOutput.writeInt(this.arrayOfCustomObjects);
-    this.arrayOfCustomObjects.forEach(function(element) {
-        dataOutput.writeObject(element);
-    });
-    ...
+function Address(street, zipCode, city, state) {
+    this.street = street;
+    this.zipCode = zipCode;
+    this.city = city;
+    this.state = state;
 }
 
-readData(dataInput) {
-    ...
-    var arrayOfCustomObjects = [];
-    var lenOfArray = dataInput.readInt();
-    for (i=0;i<lenOfArray;i++) {
-        arrayOfCustomObjects.push(dataInput.readObject());
-    }
-    this.arrayOfCustomObjects = arrayOfCustomObjects;
-    ...
-}
+Address.prototype.getClassId = function () {
+    return 1;
+};
+
+Address.prototype.getFactoryId = function () {
+    return 1;
+};
+
+Address.prototype.writeData = function (objectDataOutput) {
+    objectDataOutput.writeUTF(this.street);
+    objectDataOutput.writeInt(this.zipCode);
+    objectDataOutput.writeUTF(this.city);
+    objectDataOutput.writeUTF(this.state);
+};
+
+Address.prototype.readData = function (objectDataInput) {
+    this.street = objectDataInput.readUTF();
+    this.zipCode = objectDataInput.readInt();
+    this.city = objectDataInput.readUTF();
+    this.state = objectDataInput.readUTF();
+};
 ```
 
+IdentifiedDataSerializable uses `getClassId()` and `getFactoryId()` to reconstitue the object. To complete the implementation `IdentifiedDataSerializableFactory` should also be implemented and registered into `SerializationConfig` which can be accessed from `Config.serializationConfig`. The Factories responsibility is to return an instance of the right `IdentifiedDataSerializable` object, given the class id. 
```

Comment:
..reconstitute the object.


...The factory's responsibility is....
(or) 
...`IdentifiedDataSerializableFactory`s responsibility is...